cpp/ql/src/Security/CWE/CWE-190/Bounded.qll

```diff
@@ -24,6 +24,10 @@ private predicate boundedBitwiseAnd(Expr e, Expr andExpr, Expr operand1, Expr op
  * operation that may greatly reduce the range of possible values.
  */
 predicate bounded(Expr e) {
+  // There can be two separate reasons for `convertedExprMightOverflow` not holding:
+  // 1. `e` really cannot overflow.
+  // 2. `e` isn't analyzable.
+  // If we didn't rule out case 2 we would declare anything that isn't analyzable as bounded.
```

Comment:
Note that it's a pretty common pattern to have three cases to think about (1) a thing holds (2) the thing doesn't hold and (3) we weren't able to determine either.  If we had another predicate, `convertedExprNeverOverflows`, this would not be the inverse of `convertedExprMightOverflow` (due to case 3) and the choice between using `convertedExprNeverOverflows` or `not convertedExprMightOverflow` as we do here would be meaningful.